custom/enikshay/management/commands/investigations_reconciliation.py

```diff
@@ -0,0 +1,258 @@
+            coalesced_value = self.get_coalesced_value_for_case_property(
+                case_property, values
+            )
+            coalesced_values[case_property] = coalesced_value
+
+        # write values for each investigation
+        for investigation_case_id, investigation_case_values in properties_by_investigation.items():
+            other_investigation_case_values = investigation_case_values.copy()
+            other_investigation_case_values['investigation_case_id'] = investigation_case_id
+            other_investigation_case_values['modified_on'] = investigation_case_modified_on[investigation_case_id]
+            other_investigation_case_values['episode_id'] = episode_case_id
+            other_investigation_case_values['update_or_close'] = (
+                'to_be_updated'
+                if investigation_case_id == retain_case_id
+                else 'to_be_closed'
+            )
+            self.writerow(other_investigation_case_values)
+
+        # write final coalesced values
+        coalesced_investigation_case = coalesced_values.copy()
+        coalesced_investigation_case['investigation_case_id'] = retain_case_id
+        coalesced_investigation_case['modified_on'] = latest_investigation_case.modified_on
+        coalesced_investigation_case['episode_id'] = episode_case_id
+        coalesced_investigation_case['update_or_close'] = 'update'
+        self.writerow(coalesced_investigation_case)
+
+    def close_or_update_investigation_cases(self, all_cases, retain_case_id, episode_case_id,
+                                            investigation_interval, updates):
+        all_case_ids = [investigation_case.case_id for investigation_case in all_cases]
+        # ToDo: refetch investigation_cases in case the len of set is different from list
+        # remove duplicates in case ids to remove so that we don't retain and close
+        # the same case by mistake
+        all_case_ids = set(all_case_ids)
+        case_ids_to_close = all_case_ids.copy()
+        case_ids_to_close.remove(retain_case_id)
+        for investigation_case in all_cases:
+            self.writerow({
+                "episode_case_id": episode_case_id,
+                "investigation_interval": investigation_interval,
+                "investigation_case_id": investigation_case.case_id,
+                "modified_on": investigation_case.get_case_property(DATE_MODIFIED_FIELD),
+                "updates": updates,
+                "update/close": ('update' if investigation_case.case_id == retain_case_id
+                                 else 'closed')
+            })
+        if not self.dry_run:
+            updates = [(case_id, {'close_reason': "duplicate_reconciliation"}, True)
+                       for case_id in case_ids_to_close]
+            bulk_update_cases(DOMAIN, updates, self.__module__)
+
+    def get_coalesced_value_for_case_property(self, case_property, all_values):
+        #ToDo: write logic for this
```

Comment:
There's a ToDo here

Reply:
Ya. Need some cases to verify what should be done here but there are not any of now.